module/signature/checksum.go

```diff
@@ -0,0 +1,66 @@
+package signature
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/onflow/flow-go/model/flow"
+)
+
+// CheckSumLen is fixed to be 16 bytes, which is also md5.Size
+const CheckSumLen = 32
+
+// CheckSumFromIdentities returns checksum for the given identities
+func CheckSumFromIdentities(identities []flow.Identifier) [CheckSumLen]byte {
+	return flow.MakeID(EncodeIdentities(identities))
```

Comment:
I tried to use `crypto/md5.Sum` method, but the linter complains it's too weak, so I switched to sha3.

Reply:
I'm still working on the test cases. But I would like to get some opinions on the choice of
1. using `flow.MakeID` to compute the check sum
2. encodes identities by concatenating the bytes.

@AlexHentschel @tarakby

Reply:
To confirm the use-case of the checksum, is this only a measure to find honest bugs? Or are we considering malicious behaviours when a malicious QC would target a collision (find 2 list of identities that have the same checksum, so that the receiver doesn't catch the inconsistency) ?

Reply:
I'll continue the review depending on the answer to the question :)

Reply:
The checksum is for debugging. 

Even if such collision (2 list of identifiers having the same checksum) exists, the receiver will still reject the malicious QC because the signature will not match with the signer. And the block will be rejected. This is not supposed to happen in a non-BFT environment, except for software bug, in which case, the checksum will be used to debug. 

That being said, this checksum is only useful in non-BFT environment for debugging purpose. In a BFT environment, the checksum is not very useful.

Reply:
The encoding into bytes (point 2.) is fine 👌🏼